src/main/java/storage/Storage.java

```diff
@@ -0,0 +1,100 @@
+package storage;
+
+import command.Commands;
+import dukeExceptions.DukeException;
+import parser.Parser;
+import task.ListOfTask;
+import task.Task;
+import ui.Ui;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Storage {
+
+    /**
+     * Saves the list to a specific file.
+     *
+     * @param listOfTask The task list to save.
+     */
+    public static void save(ArrayList<Task> listOfTask) {
+        File writeData = new File("./src/data/duke.txt");
+        try {
+            writeData.createNewFile();
+            FileWriter writer = new FileWriter(writeData);
+            listOfTask.forEach(x-> {
+                try {
+                    writer.write(x.write());
+                    if (x.getIsDone()) {
+                        writer.write("mark " + (listOfTask.indexOf(x) + 1) + "\n");
+                    }
+                } catch (IOException e) {
+                    System.out.println("You do not have access to write to your save file");
+                }

```

Comment:
This try catch might be a bit redundant since it is enclosed by a try catch that catches the same exception.

Reply:
Um, this part here is actually a lamda expression.
So its like a class in itself?
Its like a consumer class?
That means that I need to catch that exception, or at least thats what my IDE said.

Reply:
Unless I declare that the class throws a exception?